### PR TITLE
chore(message): remove classic theme support - FE-3115/FE-3142

### LINKED ISSUE #3203: Message "title" prop invalid type
### Current behaviour
If you provide a component instead of a string as a title, a warning is thrown in the console:
```
Warning: Failed prop type: Invalid prop `title` of type `object` supplied to `MessageContent`, expected `string`.
    in MessageContent (created by Message)
```

### Expected behaviour
It shouldn't throw this warning, docs mention that title should be of type `node` and propTypes of the Message component are correct:
```

Message.propTypes = {
  /** set type of message based on new DLS standard */
  variant: _propTypes.default.oneOf(_optionsHelper.default.colors),

  /** set type of message. This is legacy property */
  as: _propTypes.default.string,

  /** set border to component */
  border: _propTypes.default.bool,

  /** set content to component */
  children: _propTypes.default.node,

  /** set custom class to component */
  className: _propTypes.default.string,

  /** set custom id to component root */
  id: _propTypes.default.string,

  /** show message component */
  open: _propTypes.default.bool,

  /** function runs when user click dismiss button */
  onDismiss: _propTypes.default.func,

  /** set corners to be rounded or sharp */
  roundedCorners: _propTypes.default.bool,

  /** set message title */
  title: _propTypes.default.node,

  /** set background to be invisible */
  transparent: _propTypes.default.bool,

  /** Determines if the close icon is shown */
  showCloseIcon: _propTypes.default.bool
};
```
### Reproducible example
https://codesandbox.io/s/vibrant-blackburn-67f7d?file=/src/index.js

### Additional context
image

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| carbon-react     | 40.0.0
| carbon-factory   | - 
| react-scripts    | 3.4.3
| React            | 16.13.1
| Browser          | Chrome 85.0.4183.83 (Official Build) (64-bit)
| npm              | yarn@1.22.5
| Operating System | MacOS 10.15.6

### PR DESCRIPTION
**Pull request by mkrds**
BREAKING CHANGE: message component no longer supports classic theme

This PR removes classic theme support from `Message` component

This PR also fixes FE-3142
Closes #3203
### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### CODE REVIEW

**Review by DipperTheDan — APPROVED** [review]

File: src/components/message/message.spec.js [hunk 1]
```diff
@@ -86,93 +86,14 @@ describe("Message", () => {
...
-
   describe("when closeIcon is not provided", () => {
     let wrapper, onDismissCallback;
 
     beforeEach(() => {
       onDismissCallback = jest.fn();
       wrapper = shallow(
         <Message
-          theme={classicTheme}
+          // theme={classicTheme}
```
Comment (jamime) [inline src/components/message/message.spec.js:96]: Please remove commented code

File: src/components/message/message.spec.js [hunk 2]
```diff
@@ -209,7 +130,7 @@ describe("Message", () => {
...
```
Comment (jamime) [inline src/components/message/message.spec.js:133]: Please remove commented code

**Review by jamime — CHANGES_REQUESTED** [review]
Please can you add details about this codemod in the commit so it is included in the release note.

https://github.com/Sage/carbon-codemod/tree/master/transforms/message-remove-classic-theme

**Review by DipperTheDan — APPROVED** [review]

**Review by jamime — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by Alec-Walker-Sage — APPROVED** [review]

**Pull request merged**